third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -214,10 +214,26 @@ impl LazyLoadedFunction {
             .runtime_environment()
             .vm_config()
             .ty_builder;
-
+        let enable_capture_option = layout_converter
+            .runtime_environment()
+            .vm_config()
+            .enable_capture_option;
         mask.extract(fun.param_tys(), true)
             .into_iter()
             .map(|ty| {
+                if !enable_capture_option {
```

Comment:
I think we should only do this in `layout_converter`: when constructing a layout, not allow option types (probably need one more entry point) - this way we 1) do not re-implement traversal 2) do not traverse type 2 times for captured args + re-substitute types and re-access closure of modules for this type, 3) we also have bounds on layout construction + metering, current traversal can make things DOSable / hard to remove later - there are no limits on recursion depths here

Reply:
I see, then we need to propagate a flag to tell the layout_converter to only check option during construction of  captured layout?